HLTrigger/HLTcore/interface/TriggerExpressionParser.h

```diff
@@ -37,16 +48,24 @@ namespace triggerExpression {
 
       operand %= (prescale | element);
 
-      unary = (operand[qi::_val = qi::_1] | (qi::lit("NOT") >> operand)[qi::_val = new_<OperatorNot>(qi::_1)]);
+      unary = ((operand_not >> unary)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
```

Comment:
As pointed out by Andrea B., another solution for the "NOT NOT" case is to make the parser simplify expressions discarding even occurrences of "NOT" (e.g. reducing "NOT NOT NOT" to "NOT"), as follows:
```diff
-     unary = ((operand_not >> operand)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
+     unary = *(operand_not >> operand_not) >>
              ((operand_not >> operand)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
```
I haven't applied this in this PR, because I have a slight preference for making the parser return exactly the expression it is given (a bit dumber, but a bit more transparent, imho), as unlikely as it is to use multiple NOTs in a row.